lib/session_encryptor.rb

```diff
@@ -173,15 +173,20 @@ def alert_or_raise_if_contains_sensitive_value!(string, hash)
   def alert_or_raise_if_contains_sensitive_keys!(hash)
     hash.deep_transform_keys do |key|
       if SENSITIVE_KEYS.include?(key.to_s)
-        exception = SensitiveKeyError.new("#{key} unexpectedly appeared in session")
-        if IdentityConfig.store.session_encryptor_alert_enabled
-          NewRelic::Agent.notice_error(
-            exception, custom_params: {
-              session_structure: hash.deep_transform_values { |_v| '' },
-            }
-          )
-        else
-          raise exception
+        unless hash['warden.user.user.session']&.dig('pii_like_key').
+            present? && hash['warden.user.user.session']['pii_like_key'].any? do |key|
+              hash['warden.user.user.session'].key?(key)
+            end
```

Comment:
I don't think this works the way we'd want it to work. Looking at the code, this would allow an actual sensitive value to exist in the session, so long as an exempted `pii_like_key` _also_ exists in the session.

Reply:
Understood. Refactored to incorporate suggestion here: https://github.com/18F/identity-idp/pull/11293#discussion_r1806504883
Adding a test to cover a different auth method that finally succeeds.